lib/mutant/isolation.rb

```diff
@@ -33,30 +37,41 @@ module Fork
       #   if block terminates abnormal
       #
       # rubocop:disable MethodLength
-      def self.call(&block)
-        IO.pipe(binmode: true) do |reader, writer|
-          writer.binmode
+      def call(&block)
+        io.pipe(binmode: true) do |reader, writer|
           begin
```

Comment:
The object looks horrible, mostly because we have to pass down the IO globals we captured as state. Once I finished wraping these up into `Mutant::IO` we can revisit this for now we'll go with the data clump.